### ISSUE TITLE
Unable to set a gap (or border) on a motion path or rect

### ISSUE THREAD

**Issue by oscarginette** [airbnb/visx#1529]
Due to the WCAG (Accessibility) rules, there is a need to have a gap between stacked shapes. In this case, areas and stacks need to have a gap between them.

After researching the possibilities, no easy solution has been found. There is no option on visx to add a gap on top or bottom (or sides) to a path or rect.

This is how the chart should look:
<img width="1021" alt="Screenshot 2022-06-21 at 17 54 02" src="https://user-images.githubusercontent.com/49130842/174844014-a9a1e7cb-64f8-4d01-a99b-a09fd57dca90.png">

We would love to be able to set a gap on top, bottom, left, or right where the shape is not displaced. It would work similarly to the padding in CSS.

Thank you,

Oscar

**Comment by hshoff** [comment]
Hi Oscar, could you use `stroke` and `strokeWidth` together to add a border and simulate the gap?

**Comment by oscarginette** [comment]
> Hi Oscar, could you use `stroke` and `strokeWidth` together to add a border and simulate the gap?

Hi Harrison, thanks for answering. The 'stroke' and 'strokeWidth' solution was our first approach. Nonetheless, it does not work for us since it "adds a gap" in the whole shape. border. We are not able to just set a gap only on the top or bottom of these shapes.

We will be more than happy to hear if there is any other idea we have not thought of yet.

**Comment by hshoff** [comment]
This can be accomplished by rendering toplines (or baselines) on top of the stack areas. To do this, map over the stack points `[[lowerpoint, upperpoint],...]` to get `[upperpoint, ...]`. Then render a `<LinePath />` for each topline with a stroke width for the gap size. Can set the stroke color to match the background.

![image](https://user-images.githubusercontent.com/339208/175967917-5b88cbaa-1b8c-4515-8f13-1735a1165d02.png)
Demo: https://codesandbox.io/s/adoring-heisenberg-6y19s5?file=/Example.tsx:2246-4061


```tsx
<AreaStack<CityTemperature, CityName>
  data={data}
  keys={keys}
  x={(d) => dateScale(parseDate(d.data.date))}
  y0={(d) => temperatureScale(d[0])}
  y1={(d) => temperatureScale(d[1])}
>
  {({ stacks, path }) => {
    return (
      <>
        {/* render stack areas underneath gap lines */}
        {stacks.map((stack) => {
          return (
            <path
              key={`stack-${stack.key}`}
              d={path(stack) || ""}
              stroke="transparent"
              fill={colorScale(stack.key)}
            />
          );
        })}
        {/* render gap lines ontop of stack areas */}
        {stacks.map((stack, index) => {
          // skip topline for top area
          if (index === stacks.length - 1) {
            return null;
          }
          // stack = [[lowerpoint, upperpoint], ...]
          // topline is line of upperpoints
          const topline = stack.map((d) => ({
            date: d.data.date,
            temperature: d[1]
          }));
          return (
            <LinePath
              data={topline}
              x={(d) => dateScale(parseDate(d.date))}
              y={(d) => temperatureScale(d.temperature)}
              stroke={background}
              strokeWidth={STACK_GAP}
              strokeLinecap="butt"
            />
          );
        })}
      </>
    );
  }}
</AreaStack>
```

**Comment by williaster** [comment]
Just noting that `d3-shape` itself doesn't expose a way to create gaps between stack elements ([docs](https://github.com/d3/d3-shape#stacks)), my guess would be because it might be misleading visually. So this is more of a limitation in `d3` than `visx`.
